io-engine/src/bdev/nexus/nexus_io.rs

```diff
@@ -301,23 +300,74 @@ impl<'n> NexusBio<'n> {
                         IoSubmissionFailure::Write,
                     ),
                 );
-
-                self.fail();
+                r
             } else {
                 self.ctx_mut().in_flight = 1;
+                r
             }
-            r
         } else {
             error!(
                 "{:?}: read I/O submission failed: no children available",
                 self
             );
 
-            self.fail();
             Err(CoreError::NoDevicesAvailable {})
         }
     }
 
+    /// Submit a read operation to the next suitable replica.
+    /// In case of submission error the requiest is transparently resubmitted
+    /// to the next available replica.
+    fn do_readv(&mut self) -> Result<(), CoreError> {
+        match self.__do_readv_one() {
```

Comment:
nit: might be able to reduce nesting by doing:
```rust
match self.__do_readv_one() {
    Err(CoreError::NoDevicesAvailable {}) => {
       self.fail();
       Err(e)
    }
    Err(_) => { ... }
    Ok(_) => Ok(()),
}
```

Reply:
Yup, could be done.
Linter didn't complain, thought. :-)